### ISSUE TITLE
improvements to ft_dipolesimulation

### ISSUE THREAD

**Issue by schoffelen** [fieldtrip/fieldtrip#2011]
During the MEEG analysis meeting @mcpiastra and @vitpia presented a use case for simulations with:
- precomputed (e.g. FEM) leadfields)
- more realistic noise

I think this is a use case that shouldn't be too difficult to build in.

**Comment by schoffelen** [comment]
Note: the more realistic noise can either consist of pink noise, or noise with a prespecified non-diagonal covariance, or it could be taken from real data.

**Comment by robertoostenveld** [comment]
I think it would be nice to combine ft_dipolesimulation with ft_timelocksimulation or ft_freqsimulation, where the latter two would generate the time series that get mapped through the lead fields.

**Comment by schoffelen** [comment]
oh no. :)

**Comment by schoffelen** [comment]
and then we also have ft_connectivitysimulation....

I am wondering whether for the purpose of @mcpiastra and @vitpia we should just do the forward modelling by hand...

**Comment by schoffelen** [comment]
I think it would be nice to segregate the simulation of the temporal stuff from the simulation of the spatial stuff. This would imply a somewhat larger change to the code with potential loss of backward compatibility, so would need some thought and discussion before going ahead I'd say.

**Comment by robertoostenveld** [comment]
right now it allows the specification of the source activity in 

```
% The timecourse of the dipole activity is given as a cell-array with one
% dipole signal per trial
%   cfg.dip.signal     = cell-array with one dipole signal per trial
% or by specifying the parameters of a sine-wave signal
%   cfg.dip.frequency  =   in Hz
%   cfg.dip.phase      =   in radians
%   cfg.dip.amplitude  =   per dipole
```

Considering that the source activity time courses are computed externally, I think it would make sense to do it like this

```
cfg = ...
sourceactivity = ft_timelocksimulation(cfg)

cfg = ...
cfg.headmodel = ...
forwardsimulation = ft_dipolesimulation(cfg,  sourceactivity)
```

**Comment by schoffelen** [comment]
I agree. Yet, in ft_timelocksimulation the functionality to generate the time series is somewhat different from the way the time series are created (and parametrized) in ft_dipolesimulation. And then there's also ft_connectivitysimulation ...

**Comment by robertoostenveld** [comment]
If we make sure that the specification of cfg.dip.xxx for the "parameters of a sine-wave signal" also gets supported by `fr_freqsimulation`, they would play nicely side-by-side. 

Another consideration that I have had in the past is that the output of `ft_dipolefitting` and possibly also of `ft_sourceanalysis` with cfg.method=mne should be forward-computable again.

**Comment by robertoostenveld** [comment]
it would be good to have a mock-up example script that demonstrates how we would like it to behave, and to make sure that that sufficiently covers the desired use-cases from @vitpia and @mcpiastra

**Comment by vitpia** [comment]
Ok, I will prepare such a script. There are two separate issues: 
1) parameters of the signal and what kind of noise is added
2) ft_dipolesimulation now recomputes the leadfields (lines 240 and 242). Following issue #1377 , the flexibility to provide pre-computed leadfields was implemented for ft_sourceanalysis but not for ft_dipolesimulation.

**Comment by vitpia** [comment]
I was thinking about 1. more. I am not sure how meaningful it is to have total control of the type of noise added beyond what is provided now. If the type of noise matters, then estimating noise from data may make more sense than just pink noise (which is what the code does now). Suggested by @schoffelen: add resting-state like data as noise. Would it make sense to allow the user to determine a period of pre-stimulus baseline to be used as resting-state noise? Then what I'm using now in my simulations (trial-level baseline is the trial-level "noise", which is added to the time course of the dipole signal, which is fixed).

but I'm not sure if this proposal is only addressing the ft_dipolesimulation case, whereas you wanted it to be more applicable?

**Comment by schoffelen** [comment]
Just to add to this, as a placeholder to remember, in an interaction with Teresa Cheung, we observed that there is some confusion with respect to the use (and expected functionality) of cfg.chanunit/cfg.dipoleunit. 

General question to @robertoostenveld: according to FT-philosophy - if stuff is defined in SI-units - the output units are SI and correct. This refers to units being m/V/T etc, without milli/centi/mirco/femto, right? What is then the assumed default dipoleunit: should that be Am? The docs seem to suggest nAm, which is why I ask...

**Comment by robertoostenveld** [comment]
yes, dipole strength in SI units is Am, and volume conductivity is in S/m (and "volume resistivity" is one-over-volume-conductivity and is in Ωm). The doc is then incorrect. 

Note that volume conductivity (-> larger value, more conductive) is often not specified by the user but taken from literature and in SI units. Hence when people mix their head models (in cm or mm) with literature conductivities (in S/m), units are inconsistent.  See also https://www.fieldtriptoolbox.org/faq/what_is_the_conductivity_of_the_brain_csf_skull_and_skin_tissue/

**Comment by robertoostenveld** [comment]
```
>> ft_scalingfactor('A*m', 'nA*m')
ans =
   1.0000e+09
```
